File: SW Expert Academy/D2/01959. 두 개의 숫자열/두 개의 숫자열.py
for tc in range(int(input())):
    N,M = map(int, input().split())
    A = list(map(int, input().split()))
    B = list(map(int, input().split()))

    print("#%d"%(tc+1), end=' ')

    if N > M:
        A,B = B,A
        N,M = M,N

    max_sum = -1
    for i in range(M-N+1):
        sum = 0
        for j in range(N):
            sum += A[j]*B[i+j]

        max_sum = max(max_sum, sum)

    print(max_sum)

File: SW Expert Academy/D2/01959. 두 개의 숫자열/두 개의 숫자열.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
	int T; cin>>T;
	for(int tc=0; tc<T; tc++){
		int N,M; cin>>N>>M;
		vector<int> A(N), B(M);
		
		for(int i=0; i<N; i++) cin>>A[i];
		for(int i=0; i<M; i++) cin>>B[i];
		
		printf("#%d ", tc+1);
		
		// A의 길이를 더 짧게 맞춘다
		if(N > M){
			vector<int> tmp = A;
			A = B;
			B = tmp;
			
			int n = N;
			N = M;
			M = n;
		}
		
		// B의 0번째부터 하나씩 A를 대가며 계산하기
		int max = -1;
		for(int i=0; i<=M-N; i++){
			int sum = 0;
			for(int j=0; j<N; j++)
				sum += A[j]*B[i+j];
			
			if(sum > max) max = sum;
		}
		
		printf("%d\n", max);
	}
}
